src/main/java/org/rumbledb/server/RumbleHandler.java

```diff
@@ -0,0 +1,203 @@
+
+    private enum StatusCode {
+        SUCCESS(200),
+        SERVER_ERROR(500);
+
+        private int code;
+
+        StatusCode(int code) {
+            this.code = code;
+        }
+
+        public int getCode() {
+            return this.code;
+        }
+    }
+
+    public RumbleHandler() {
+    }
+
+    private void sendResponse(HttpExchange exchange, StatusCode code, String response) throws IOException {
+        exchange.sendResponseHeaders(code.getCode(), response.getBytes().length);
+        OutputStream stream = exchange.getResponseBody();
+        stream.write(response.getBytes());
+        stream.close();
+    }
+
+    public String[] getCLIArguments(String query) throws UnsupportedEncodingException {
+        Map<String, String> queryParameters = new HashMap<String, String>();
+        if (query == null) {
+            query = "";
+        }
+        for (String pair : query.split("&")) {
+            int index = pair.indexOf("=");
+            if (index != -1) {
+                queryParameters.put(
+                    URLDecoder.decode(pair.substring(0, index), "UTF-8"),
+                    URLDecoder.decode(pair.substring(index + 1), "UTF-8")
+                );
+            }
+        }
+        String[] args = new String[queryParameters.keySet().size() * 2];
+        int i = 0;
+        for (String param : queryParameters.keySet()) {
+            args[i++] = "--" + param;
+            args[i++] = queryParameters.get(param);
+        }
+        return args;
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            URI uri = exchange.getRequestURI();
+            String queryString = uri.getQuery();
+            String args[] = getCLIArguments(queryString);
+
+            RumbleRuntimeConfiguration configuration = new RumbleRuntimeConfiguration(args);
+            SparkSessionManager.COLLECT_ITEM_LIMIT = configuration.getResultSizeCap();
+
+            JsoniqQueryExecutor translator = new JsoniqQueryExecutor(configuration);
+            List<Item> items;
+            if (configuration.getQueryPath() != null) {
+                if (configuration.getOutputPath() != null && !exchange.getRequestMethod().equals("POST")) {
+                    throw new CliException(
+                            "The POST method must be used if specifying an output path, as this has side effects."
+                    );
+                }
+                items = translator.runQuery(
+                    configuration.getQueryPath(),
+                    configuration.getOutputPath()
+                );
+            } else {
+                if (configuration.getOutputPath() != null && !exchange.getRequestMethod().equals("POST")) {
+                    throw new CliException(
+                            "The POST method must be used if specifying an output path, as this has side effects."
+                    );
+                }
```

Comment:
This exception seems duplicated in the if and else statements. Can we take move it outside and remove duplication?

Reply:
I believe this comment remained unaddressed